core/trino-main/src/main/java/io/trino/sql/analyzer/Analysis.java

```diff
@@ -832,6 +845,48 @@ public void addEmptyColumnReferencesForTable(AccessControl accessControl, Identi
         tableColumnReferences.computeIfAbsent(accessControlInfo, k -> new LinkedHashMap<>()).computeIfAbsent(table, k -> new HashSet<>());
     }
 
+    public void addLabelDereferences(Map<NodeRef<DereferenceExpression>, LabelPrefixedReference> dereferences)
+    {
+        labelDereferences.putAll(dereferences);
+    }
+
+    public LabelPrefixedReference getLabelDereference(DereferenceExpression expression)
+    {
+        return labelDereferences.get(NodeRef.of(expression));
+    }
+
+    public void addPatternRecognitionFunctions(Set<NodeRef<FunctionCall>> functions)
+    {
+        patternRecognitionFunctions.addAll(functions);
+    }
+
+    public boolean isPatternRecognitionFunction(FunctionCall functionCall)
+    {
+        return patternRecognitionFunctions.contains(NodeRef.of(functionCall));
+    }
+
+    public void setRange(RangeQuantifier quantifier, Range range)
+    {
+        ranges.put(NodeRef.of(quantifier), range);
+    }
+
+    public Range getRange(RangeQuantifier quantifier)
+    {
+        Range range = ranges.get(NodeRef.of(quantifier));
+        return requireNonNull(range, "missing range for quantifier " + quantifier);
```

Comment:
Split into check and return. This for makes it harder to see that what's being returned is `range`